File: app/src/main/java/com/wilderpereira/lmgtfygen/dagger/module/PresenterModule.java
package com.wilderpereira.lmgtfygen.dagger.module;

import com.wilderpereira.lmgtfygen.presentation.MainContract;
import com.wilderpereira.lmgtfygen.presentation.MainPresenter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Wilder on 31/01/17.
 */

@Module
public class PresenterModule {

    @Singleton
    @Provides
    MainContract.Presenter providesPresenter(){
        return new MainPresenter();
    }
}

File: app/src/main/java/com/wilderpereira/lmgtfygen/dagger/module/PresenterModule.kt
package com.wilderpereira.lmgtfygen.dagger.module

import com.wilderpereira.lmgtfygen.presentation.MainContract
import com.wilderpereira.lmgtfygen.presentation.MainPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by Wilder on 31/01/17.
 */

@Module
open class PresenterModule {

    @Singleton
    @Provides
    internal fun providesPresenter(mainPresenter: MainPresenter): MainContract.Presenter = mainPresenter
}
